### PR TITLE
scx: 1.0.5 -> 1.0.6; build all rust subpackages together

### PR DESCRIPTION
**Pull request by JohnRTitor**
Changelog: https://github.com/sched-ext/scx/compare/v1.0.5...v1.0.6

### Why the rewrite?

Previous implementation had Cargo.locks all over the place, against the wishes of https://github.com/NixOS/nixpkgs/issues/327063

Now we can just build the rust subpackages from the workspace. Moreover, previous implementation was incompatible with v1.0.6, Rust binaries segfaulted (https://github.com/sched-ext/scx/issues/927)

### NOTE to users:

`scx_rustland` seems to have been affected by a kernel regression (https://github.com/sched-ext/scx/issues/788), please choose another scheduler if you are affected.

This PR removes subpackages like `scx.rustland`, `scx.lavd`, ... If you were using any of these packages, just change it to `scx.rustscheds`, it contains all rust based schedulers. `scx.full` is the same, will work as expected.

### NOTE to maintainers:

In future, when updating this package, run the `update.sh`, it will get the latest release version of scx, as well as get the compatible bpftool and libbpf revisions.

~Make sure to update the `cargoHash` of `scx.rustscheds`, by building the package and getting the correct hash from the build failure. I couldn't figure out how to implement auto updating this in the script.~
Update script should now also update cargoHash properly, if it does not please do it manually.

Closes https://github.com/NixOS/nixpkgs/pull/357571

###

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
@ofborg build scx.full

File: pkgs/os-specific/linux/scx/default.nix [hunk 1]
```diff
@@ -1,79 +1,36 @@
...
+      owner = "sched-ext";
+      repo = "scx";
+      rev = "refs/tags/v${versionInfo.scx.version}";
+      inherit (versionInfo.scx) hash;
+    };
+
+    meta = {
+      homepage = "https://github.com/sched-ext/scx";
+      license = lib.licenses.gpl2Only;
+      platforms = lib.platforms.linux;
```
Comment (PedroHLC) [inline pkgs/os-specific/linux/scx/default.nix:22]: add `changelog = "https://github.com/sched-ext/scx/releases/tag/v${versionInfo.scx.version}"`

**Review by PedroHLC — APPROVED** [review]
Now, that's a polished version of this package!

**Comment by JohnRTitor** [comment]
@ofborg build scx.full

File: pkgs/os-specific/linux/scx/default.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/os-specific/linux/scx/default.nix:23]: Done, thanks

**Comment by JohnRTitor** [comment]
@ofborg build scx.full

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358154`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>scx.cscheds</li>
    <li>scx.cscheds.bin</li>
    <li>scx.cscheds.dev</li>
    <li>scx.full</li>
    <li>scx.full.bin</li>
    <li>scx.full.dev</li>
    <li>scx.rustscheds</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>scx.full</li>
    <li>scx.full.bin</li>
    <li>scx.full.dev</li>
    <li>scx.rustscheds</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>scx.cscheds</li>
    <li>scx.cscheds.bin</li>
    <li>scx.cscheds.dev</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Comment by GaetanLepage** [comment]
Fails on `aarch64-linux` with: https://paste.glepage.com/raw/bear-toad-sloth

**Comment by GaetanLepage** [comment]
> Yeah, can't fix the aarch64 build, it's not fully supported by upstream IIRC.

Ok, let's add it to `badPlatforms` then I guess.

**Comment by JohnRTitor** [comment]
Done, also edited the module, as `scx.rustland` is replaced by `scx.rustscheds`.

**Comment by Gliczy** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358154`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>scx.cscheds</li>
    <li>scx.cscheds.bin</li>
    <li>scx.cscheds.dev</li>
    <li>scx.full</li>
    <li>scx.full.bin</li>
    <li>scx.full.dev</li>
    <li>scx.rustscheds</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

**Comment by Gliczy** [comment]
Nice work!

I've tested a few schedulers and everything seem to run as expected.

**Pull request merged**